library/pk.c

```diff
@@ -84,7 +90,8 @@ void mbedtls_pk_restart_init( mbedtls_pk_restart_ctx *ctx )
  */
 void mbedtls_pk_restart_free( mbedtls_pk_restart_ctx *ctx )
 {
-    if( ctx == NULL || ctx->pk_info == NULL ||
+    PK_VALIDATE( ctx != NULL );
```

Comment:
It can be argued whether it's right, but we have decided to document that `xxx_free()` functions accept `NULL` parameters and behave as no-ops in this case.

Reply:
My mistake. Fixed.